src/criteriaBuilder.js

```diff
@@ -0,0 +1,69 @@
+import extend from 'extend';
+
+export class CriteriaBuilder {
+
+  updateCriteria() {
+    let blocks = [];
+
+    this.filters.forEach((block, index) => {
+      let filtersParsed = {};
+
+      block.forEach(filter => {
+        if (!this.hasValue(filter.data.value)) {
+          return;
+        }
+
+        filtersParsed = extend(filtersParsed, this.parseField(filter.data.field, this.parseOperator(filter.data)));
+      });
+
+      blocks.push(filtersParsed);
+    });
+
+    this.criteria = (blocks.length > 1) ? {where: {or: blocks}} : {where: blocks[0]};
```

Comment:
@Rw i.d suggest to generalize the format and to move the logic to orm or api. then users can install aurelia-api-hapi-adapter and be done with it for all our plugins